api/util.py

```diff
@@ -84,6 +85,20 @@ def user_perm(permissions, _id, site=None):
             return perm
     return {}
 
+def is_user_id(uid):
+    """
+    Checks to make sure uid matches uid regex
+    """
+    pattern = re.compile('^[0-9a-zA-Z.@_-]*$')
+    return bool(pattern.match(uid))
+
+def is_group_id(gid):
+    """
+    Checks to make sure uid matches uid regex
+    """
+    pattern = re.compile('^[0-9a-z][0-9a-z.@_-]{0,30}[0-9a-z]$')
+    return bool(pattern.match(gid))
```

Comment:
Could these two patterns use the same regex instances as our router?
Maybe move the regexes to this file, and import them into the router?